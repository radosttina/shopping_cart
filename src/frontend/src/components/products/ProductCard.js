import React, { Component } from "react";
import PropTypes from "prop-types";

export class ProductCard extends Component {
  render() {
    return (
      <li key={this.props.product.id} className="product_item">
        <span> {this.props.product.name} </span>
        <span> {this.props.product.price} </span>
        <img className="product_item_image" src={this.props.product.product_image} />
        <button className="add_to_cart_btn" onClick={this.props.onClick.bind(this, this.props.product.id)}>{this.props.userActionText}</button>
      </li>
    );
  }
}

ProductCard.propTypes = {
  product: PropTypes.object.isRequired
}

export default ProductCard;
