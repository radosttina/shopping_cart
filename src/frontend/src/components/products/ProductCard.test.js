import React from "react";
import ProductCard from "./ProductCard";
import renderer from "react-test-renderer";
import { render } from "../../../test-utils";

test("ProductCard should render product details.", () => {
  const product = { id: 1, name: "Product", "price": 1000.00};

  const component = render(
    <ProductCard product={product} onClick={()=>{}}></ProductCard>
  );

  let tree = component.container;
  expect(tree).toMatchSnapshot();
});