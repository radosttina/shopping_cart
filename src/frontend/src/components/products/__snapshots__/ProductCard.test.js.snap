// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render product details. 1`] = `
<div>
  <li
    class="product_item"
  >
    <span>
       
      Product
       
    </span>
    <span>
       
      1000
       
    </span>
    <img
      class="product_item_image"
    />
    <button
      class="add_to_cart_btn"
    />
  </li>
</div>
`;
